frontend/extensions/helpers.lua

```diff
@@ -0,0 +1,275 @@
+		UIManager:show(InfoMessage:new { text = message, icon_file = 'resources/info-bug.png', timeout = timeout })
+
+	else
+		UIManager:show(InfoMessage:new { text = message, icon_file = 'resources/info-bug.png', timeout = timeout, dismiss_callback = dismiss_callback })
+	end
+end
+
+function Helpers:alertDebugOnce(message, timeout, dismiss_callback)
+
+	if not self.debug_once_used then
+
+		self:alertDebug(message, timeout, dismiss_callback)
+
+		self.debug_once_used = true
+	end
+end
+
+-- use timeout = nil for second argument when calling for no timeout:
+function Helpers:alertError(message, timeout, dismiss_callback)
+
+	if dismiss_callback == nil then
+		UIManager:show(InfoMessage:new { text = message, icon_file = 'resources/info-error.png', timeout = timeout })
+
+	else
+		UIManager:show(InfoMessage:new { text = message, icon_file = 'resources/info-error.png', timeout = timeout, dismiss_callback = dismiss_callback })
+	end
+end
+
+-- use timeout = nil for second argument when calling for no timeout:
+function Helpers:alertInfo(message, timeout, dismiss_callback)
+
+	if dismiss_callback == nil then
+		UIManager:show(InfoMessage:new { text = message, timeout = timeout })
+
+	else
+		UIManager:show(InfoMessage:new { text = message, timeout = timeout, dismiss_callback = dismiss_callback })
+	end
+end
+
+-- use timeout = nil for second argument when calling for no timeout:
+function Helpers:alertNoIcon(message, timeout)
+
+	UIManager:show(InfoMessage:new { text = message, show_icon = false, timeout = timeout })
+end
+
+function Helpers:shuffle(tbl)
+	for i = #tbl, 2, -1 do
+		local j = math.random(i)
+		tbl[i], tbl[j] = tbl[j], tbl[i]
+	end
+end
+
+function Helpers:split(str, pat)
+	local t = {}  -- NOTE: use {n = 0} in Lua-5.0
+	local fpat = "(.-)" .. pat
+	local last_end = 1
+	local s, e, cap = str:find(fpat, 1)
+	while s do
+		if s ~= 1 or cap ~= "" then
+			table.insert(t, cap)
+		end
+		last_end = e + 1
+		s, e, cap = str:find(fpat, last_end)
+	end
+	if last_end <= #str then
+		cap = str:sub(last_end)
+		table.insert(t, cap)
+	end
+	return t
+end
+
+function Helpers:split_path(str)
+	return self:split(str, '[\\/]+')
+end
+
+function Helpers:table_sort_by_text_prop(subject)
+
+	table.sort(subject, function(v1, v2)
+		return v1.text < v2.text
+	end)
+end
+
+function Helpers:table_sort_alphabetically(subject)
+
+	table.sort(subject, function(v1, v2)
+		return v1 < v2
+	end)
+end
+
+function Helpers:ucfirst(str)
+	return (str:gsub("^%l", string.upper))
+end
+
+function Helpers:getMiddleDialogWidth()
+
+	local orientation = Screen:getScreenMode()
+	local iwidth = (Screen:getWidth() / 2) + 20
+
+	if orientation == 'portrait' then
+		iwidth = Screen:getWidth() - 80
+	end
+
+	return iwidth
+end
+
+function Helpers:getWideDialogWidth()
+
+	local orientation = Screen:getScreenMode()
+	local iwidth = Screen:getWidth() - 120
+
+	if orientation == 'portrait' then
+		iwidth = Screen:getWidth() - 80
+	end
+
+	return iwidth
+end
+
+function Helpers:round(x)
```

Comment:
We actually have a bunch of math helpers like that in optmath.

I'll add some comment strings in so they'll show up in the docs.